File: examples/sharing/collection-create.ts
import type { JsBaoClient } from "js-bao-wss-client";

// Create a collection, put documents in it, and share the whole set at once.
export async function createCollection(
  client: JsBaoClient,
  designDocId: string,
  specDocId: string,
) {
  // #region example
  // Create a collection and put documents in it
  const collection = await client.collections.create({ name: "Project Phoenix" });
  await client.collections.addDocument(collection.collectionId, designDocId);
  await client.collections.addDocument(collection.collectionId, specDocId);

  // One grant covers the whole set — including documents added later
  await client.collections.addMember(collection.collectionId, {
    email: "alice@example.com",
    permission: "read-write",
  });
  // #endregion example
  return collection;
}

File: examples/sharing/collection-create.swift
import JsBaoClient

// Create a collection, put documents in it, and share the whole set at once.
func createCollection(
  client: JsBaoClient,
  designDocId: String,
  specDocId: String
) async throws {
  // #region example
  // Create a collection and put documents in it
  let collection = try await client.collections.create(
    params: CreateCollectionParams(name: "Project Phoenix")
  )
  _ = try await client.collections.addDocument(collectionId: collection.collectionId, documentId: designDocId)
  _ = try await client.collections.addDocument(collectionId: collection.collectionId, documentId: specDocId)

  // One grant covers the whole set — including documents added later
  _ = try await client.collections.addMember(
    collectionId: collection.collectionId,
    params: .email("alice@example.com", permission: .readWrite)
  )
  // #endregion example
}
